src/multitask_personalization/envs/pybullet/pybullet_env.py

```diff
@@ -223,13 +227,16 @@ def set_state(self, state: PyBulletState) -> None:
         set_pose(self.cup_id, state.cup_pose, self.physics_client_id)
         for book_id, book_pose in zip(self.book_ids, state.book_poses, strict=True):
             set_pose(book_id, book_pose, self.physics_client_id)
+        self.book_descriptions = state.book_descriptions
         self.current_grasp_transform = state.grasp_transform
         obj_name_to_obj = {
             None: None,
             "cup": self.cup_id,
         }
-        for i, book_id in enumerate(self.book_ids):
-            obj_name_to_obj[f"book{i}"] = book_id
+        for book_id, book_description in zip(
+            self.book_ids, self.book_descriptions, strict=True
+        ):
+            obj_name_to_obj[book_description] = book_id
```

Comment:
this can probably be cached as an attribute for faster future lookup (in get_object_id_from_name, for example)

Reply:
that's true, but since the number is so small right now (3) i'll wait until later